R/get_entity_financing_data.R

```diff
@@ -71,27 +71,32 @@ get_entity_financing_data <- function(
       by = c("fsym_id", "date", "currency")
     ) %>%
     dplyr::left_join(fsym_company_id, by = "fsym_id") %>%
-    dplyr::inner_join(sec_entity, by = c("fsym_company_id" = "fsym_id")) %>%
-    dplyr::filter(!(is.na(.data$ff_mkt_val) & is.na(.data$ff_debt))) %>%
-    dplyr::group_by(.data$fsym_id, .data$currency) %>%
-    dplyr::filter(.data$date <= .env$data_timestamp) %>%
+    dplyr::inner_join(sec_entity, by = c(fsym_company_id = "fsym_id")) %>%
     dplyr::filter(
-      lubridate::year(.data$date) == data_timestamp_year
+      !(
+        is.na(.data[["ff_mkt_val"]]) &
+          is.na(.data[["ff_debt"]])
+      )
     ) %>%
-    dplyr::filter(.data$date == max(.data$date)) %>%
+    dplyr::group_by(.data[["fsym_id"]], .data[["currency"]]) %>%
+    dplyr::filter(.data[["date"]] <= .env[["data_timestamp"]]) %>%
+    dplyr::filter(
+      lubridate::year(.data[["date"]]) == data_timestamp_year
+    ) %>%
+    dplyr::filter(.data[["date"]] == max(.data[["date"]])) %>%
     dplyr::ungroup()
 
   logger::log_trace("Downloading entity financing data.")
   entity_financing_data <- entity_financing_data %>%
     dplyr::collect() %>%
     dplyr::mutate(
       # convert units from millions to units
-      ff_mkt_val = .data$ff_mkt_val * 1e6,
-      ff_debt = .data$ff_debt * 1e6
+      ff_mkt_val = .data[["ff_mkt_val"]] * 1e6L,
+      ff_debt = .data[["ff_debt"]] * 1e6L
     ) %>%
     dplyr::distinct()
 
   # return the entity financing data -------------------------------------------
 
-  entity_financing_data
+  return(entity_financing_data)
```

Comment:
good thing he's not here then 😆 

Looking at the [tidyverse style guide](https://style.tidyverse.org/functions.html#return), that's accurate, but there's no justification. 

I fuond [this thread](https://community.rstudio.com/t/r-style-guide-explicit-return-or-not/40025) that suggest two cases where `return` might be cuumbersome are one-liners and one-pipeline functions

```r
foo <- function(x) { x^2 } #note that this also violates tidy style, since the braces aren't on their own lines

bar <- function(.data) {
  .data %>%
    fun_1 %>%
    fun_2
}
```

but that doesn't seem much worse than 

```r
bar <- function(.data) {
  df <- .data %>%
    fun_1 %>%
    fun_2
  return(df)
}
```

given the clarity that an explicit `return` provides. I suppose there might be a performance consideration with the assignment, but I doubt it, since whatever is in memory for the pipeline would need to be in memory anyway for assignment (and `gc`'d on exit).

Reply:
Here's Hadley's [comment](https://github.com/tidyverse/style/issues/131#issuecomment-470081819) on the matter:

> Returning the last evaluated expression is an absolutely key part of the R language — if you don't understand this you are going to find it hard to understand much code in the wild. Since you have to know this, I think it's much more important to reserve return() to signal early returns for exceptional conditions.

Reply:
I've seen his justification and could never fully wrap head around it. Sure, great to be explicit with early returns, but that and being explicit with final returns are not mutually exclusive

Reply:
NIT/ NB/ etc.:
I am starting (personally) to lean towards not using pipes in my production package code, as it makes debugging a bit of a pain in the butt. 
Calling `debugonce(function)` on a `function` that does:
``` r
data |>
   do_something() |>
   do_something_else()
```
is tricker to step through than:
``` r
data <- do_something(data)
data <- do_something_else(data)
return(data)
```

I have no intention of starting the `style` debate back up any-time soon ahaha, but just in case you see PRs that look like that from me, this is the reason :-)

Reply:
Yeah, I haven't seen any good way to debug a pipeline that doesn't involve actually modifying the pipeline (`debug_pipe`). `pipecleaner` looks like it might be useful, but seems to lack the tools that we would want to interrogate the data.

Reply:
Also (personally) just feels like a waste of time to try to align debugging paradigms with pipes, when it's so simple to "un-pipe" something. (Why look for a solution when one already exists)

Reply:
just FYI... personally I find piped code to be much easier to debug